### PR TITLE
Discord bot massive improvement

### PR DESCRIPTION
**Pull request by tehcoderer**
### **Added intraday ta candles with logic to grab  more days of data so the lagging indicators don't get cut off**
image
image
image

### **Before:**
image

### **After:**
image

### **etf cmds**
image
image
image
image

### **plus new table style with easy config inside config_discordbot.py**
image
image

### **sia cmds with autocomplete** 
image
image
image
image
image
image
image

### **load_candle.py helper for easy candle plotly figs** 
image

### **works with subplots as well**
image

###  **past_days. Also logic for max data intraday so users get data without an error if hitting yahoo's max days per interval** 
image
image

### **added news to interactive charts that open the articles on click**
image

### **interactive chart func helper : gets plot fig html adds js script for clickable news and sets html body background to dark**
image

### **imgur/selfhosting**
image
image

### **Before:**
image
image

### **After:**
image
image

### **added better crypto candle data api's if user adds to config_discordbot.py for 1min and volume data fixes**
image

### **Added Heikin Ashi style arg to candle based charts
image

- [ ] Summary of the change / bug fix.
- [ ] Link # issue, if applicable.
- [ ] Screenshot of the feature or the bug before/after fix, if applicable.
- [ ] Relevant motivation and context. 
- [ ] List any dependencies that are required for this change.

# How has this been tested?

* Please describe the tests that you ran to verify your changes. 
* Provide instructions so we can reproduce. 
* Please also list any relevant details for your test configuration.

# Checklist:

- [ ] Update [our Hugo documentation](https://gamestonkterminal.github.io/GamestonkTerminal/) following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/website).
- [ ] Update our tests following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/scripts).

# Others
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [ ] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

File: bots/cmds/options_cmds.py [hunk 1]
```diff
@@ -147,43 +142,72 @@ async def cc(
...
         inter: disnake.AppCmdInter,
         interval: int = commands.Param(choices=[1, 5, 15, 30, 60, 1440]),
-        past_days: int = 1,
+        past_days: int = 0,
         start="",
         end="",
+        news: bool = False,
+        heikin_candles: bool = False,
     ):
         """Display Bitcoin Chart
```
Comment (jose-donato) [inline bots/cmds/options_cmds.py:176]: missing source

Comment (tehcoderer) [reply bots/cmds/options_cmds.py]: added and pushed, Ty!

**Review by jose-donato — APPROVED** [review]
The only big issue that I see is that some commands use specific discord code as i mentioned in [issue 1464](https://github.com/GamestonkTerminal/GamestonkTerminal/issues/1464) which will make those commands fail in other platforms. 

Since this is already a huge PR and everything else looks good to me i would suggest closing this one and later have a chat to figure out the best way to move forward (i.e., make sure all commands work regardless of the platform).

Nice work!! BULLLLLLLLISH 🚀

**Comment by tehcoderer** [comment]
> The only big issue that I see is that some commands use specific discord code as i mentioned in [issue 1464](https://github.com/GamestonkTerminal/GamestonkTerminal/issues/1464) which will make those commands fail in other platforms.
> 
> Since this is already a huge PR and everything else looks good to me i would suggest closing this one and later have a chat to figure out the best way to move forward (i.e., make sure all commands work regardless of the platform).
> 
> Nice work!! BULLLLLLLLISH 🚀

The discord code doesn't matter when it comes to other bots. There's logic in helpers.py that only grabs the title/images/description dict's.

All cmds that have discord code work with other bots.


image
image
image

**Comment by jose-donato** [comment]
> > The only big issue that I see is that some commands use specific discord code as i mentioned in [issue 1464](https://github.com/GamestonkTerminal/GamestonkTerminal/issues/1464) which will make those commands fail in other platforms.
> > Since this is already a huge PR and everything else looks good to me i would suggest closing this one and later have a chat to figure out the best way to move forward (i.e., make sure all commands work regardless of the platform).
> > Nice work!! BULLLLLLLLISH 🚀
> 
> The discord code doesn't matter when it comes to other bots. There's logic in helpers.py that only grabs the title/images/description dict's.
> 
> All cmds that have discord code work with other bots.
> 
> image image image

Ooooh yea sorry!!! I was just thinking about a scenario where a user that wants to run the bot on a non-discord platform and did not have disnake installed. But poetry installs all dependencies so I guess we are okay!

File: terminal.py [hunk 2]
```diff
@@ -554,12 +554,7 @@ def log_settings() -> None:
     logger.info("SETTINGS: %s ", str(settings_dict))
 
 
-def run_scripts(
```
Comment (DidierRLopes) [inline terminal.py:557]: @tehcoderer you remove my whole PR https://github.com/GamestonkTerminal/GamestonkTerminal/pull/1455.

When adding files is better to do it manually one by one.

**Pull request merged**